core/arch/x86/decode.c

```diff
@@ -2298,6 +2323,8 @@ decode_common(dcontext_t *dcontext, byte *pc, byte *orig_pc, instr_t *instr)
     di.len = (int)(next_pc - pc);
     di.opcode = info->type; /* used for opnd_create_immed_float_for_opcode */
 
+    /* The upper half of the flags field is the evex tuple type. */
+    di.tuple_type = (byte)(info->flags >> 16);
```

Comment:
Use the named constant(s).  IMHO it would be worth limiting how many bits are used here, instead of taking all 16, just we do for the pred bits.  Then we can more easily expand the flags in the future.

Reply:
Done to both.